src/traces/bar/calc.js

```diff
@@ -51,15 +51,21 @@ module.exports = function calc(gd, trace) {
     var serieslen = Math.min(pos.length, size.length),
         cd = [];
 
-    // set position
+    // set position and size
     for(i = 0; i < serieslen; i++) {
+        var p = pos[i];
+        var s = size[i];
 
         // add bars with non-numeric sizes to calcdata
         // so that ensure that traces with gaps are
         // plotted in the correct order
 
-        if(isNumeric(pos[i])) {
-            cd.push({p: pos[i]});
+        if(isNumeric(p)) {
+            if(isNumeric(s)) {
+                cd.push({ p: p, s: s });
+            } else {
+                cd.push({ p: p });
+            }
```

Comment:
@alexcjohnson ok, you convinced me. 

With commit https://github.com/plotly/plotly.js/pull/1519/commits/0144f06bbcb46462ca7bd1187481d38d78583e83 bar calcdata has now a 1-1 correspondence with the bar `(x,y)` coordinates. Note that non-numeric bar items are cleaned up in [`bar/plot.js`](https://github.com/plotly/plotly.js/blob/a41b67670db6c35edf8ec901e0692e0eaf3c3cd6/src/traces/bar/plot.js#L82-L87) and `makeCalcdata` (called above the `serieslen` loop in `bar/calc.js`) sets non-numeric values to `BADNUM`.

---

This a pretty big change that might lead to regressions, so this PR will make be part of the next minor release `v1.26.0` only.